plugin/src/main/java/org/opensearch/ml/sdkclient/XContentClient.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.sdkclient;
+
+import static org.opensearch.action.DocWriteResponse.Result.CREATED;
+import static org.opensearch.action.DocWriteResponse.Result.DELETED;
+import static org.opensearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE;
+import static org.opensearch.common.xcontent.json.JsonXContent.jsonXContent;
+import static org.opensearch.core.xcontent.ToXContent.EMPTY_PARAMS;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.opensearch.action.delete.DeleteRequest;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.client.Client;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.sdk.DeleteCustomRequest;
+import org.opensearch.sdk.DeleteCustomResponse;
+import org.opensearch.sdk.GetCustomRequest;
+import org.opensearch.sdk.GetCustomResponse;
+import org.opensearch.sdk.PutCustomRequest;
+import org.opensearch.sdk.PutCustomResponse;
+import org.opensearch.sdk.SdkClient;
+
+public class XContentClient implements SdkClient {
+
+    private final Client client;
+    private final NamedXContentRegistry xContentRegistry;
+
+    public XContentClient(Client client, NamedXContentRegistry xContentRegistry) {
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    public CompletionStage<PutCustomResponse> putCustom(PutCustomRequest request) {
```

Comment:
What about `Update`?

Reply:
Coming soon to a PR near you, but prioritizing initial APIs to support multitenancy work.